docs/build.md

```diff
@@ -0,0 +1,225 @@
+- [Build Controller](#build-controller)
+- [Configuring a Build](#configuring-a-build)
+  - [Defining the Source](#defining-the-source)
+  - [Defining the Strategy](#defining-the-strategy)
+  - [Defining the Builder or Dockerfile](#defining-the-builder-or-dockerfile)
+  - [Defining Resources](#defining-resources)
+  - [Defining the Output](#defining-the-output)
+
+## Overview
+
+A `Build` resource allows the user to define:
+
+- source
+- strategy
+- builder
+- dockerfile
+- resources
+- output
+
+A `Build` is available within a namespace.
+
+## Build Controller
+
+The controller watches for:
+
+- Updates on the `Build` resource (_CRD instance_)
+
+When the controller reconciles it:
+
+- Validates if the referenced `StrategyRef` exists.
+- Validates if the container `registry` output secret exists.
+
+## Configuring a Build
+
+The `Build` definition supports the following fields:
+
+- Required:
+  - [`apiVersion`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Specifies the API version, for example `build.dev/v1alpha1`.
+  - [`kind`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Specifies the Kind type, for example `Build`.
+  - [`metadata`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Metadata that identify the CRD instance, for example the name of the `Build`.
+  - `spec.source.URL` - Refers to the Git repository containing the source code.
+  - `spec.strategy` - Refers to the `BuildStrategy` to be used, see the [examples](../samples/buildstrategy)
+  - `spec.builder.image` - Refers to the image containing the build tools to build the source code. (_Use this path for Dockerless strategies_)
+  - `spec.dockerfile` - Path to a Dockerfile to be used for building an image. (_Use this path for strategies that require a Dockerfile_)
+  - `spec.output`- Refers to the location where the generated image would be pushed.
+  - `spec.output.credentials.name`- Reference an existing secret to get access to the container registry.
+
+- Optional:
+  - `spec.resources` - Refers to the compute [resources](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/) used on the container where the image is built.
+  - `spec.parameters` - Refers to a list of `name-value` that could be used to loosely type parameters in the `BuildStrategy`.
+
+### Defining the Source
+
+A `Build` resource can specify a Git source, together with other parameters like:
+
+- `source.credentials.name` - For private repositories, the name is a reference to an existing secret on the same namespace containing the require `ssh` data.
+- `source.revision` - An specific revision to select from the source repository, this can be a commit or branch name.
+- `source.contextDir` - Refers to the location of the `Dockerfile`. It can be used if `spec.dockerfile` is specified.
```

Comment:
We should discuss this. `Location of the Dockerfile` is very specific. I think that it is a general capability of having a git repository, but only building an image from a source in a directory within that repository - and not a capability specific to Dockerfile based builds. Based on that, I would write:

`source.contextDir` - For repositories where your source code is in a sub-directory and not in its root, you can specify this path here. Currently only supported by the `buildah` and `kaniko` build strategies.

Reply:
yes, agree, done.